lms/djangoapps/django_comment_client/utils.py

```diff
@@ -184,19 +185,24 @@ def get_discussion_category_map(course):
                 node[level]["start_date"] = category_start_date
 
         for entry in entries:
-            node[level]["entries"][entry["title"]] = {"id": entry["id"],
-                                                      "sort_key": entry["sort_key"],
-                                                      "start_date": entry["start_date"],
-                                                      "is_cohorted": is_course_cohorted}
+            node[level]["entries"][entry["title"]] = {
+                "id": entry["id"],
+                "sort_key": entry["sort_key"],
+                "start_date": entry["start_date"],
+                "is_cohorted": course_cohort_settings.is_cohorted
+            }
 
     # TODO.  BUG! : course location is not unique across multiple course runs!
     # (I think Kevin already noticed this)  Need to send course_id with requests, store it
     # in the backend.
     for topic, entry in course.discussion_topics.items():
-        category_map['entries'][topic] = {"id": entry["id"],
-                                          "sort_key": entry.get("sort_key", topic),
-                                          "start_date": datetime.now(UTC()),
-                                          "is_cohorted": is_course_cohorted and entry["id"] in cohorted_discussion_ids}
+        category_map['entries'][topic] = {
+            "id": entry["id"],
+            "sort_key": entry.get("sort_key", topic),
+            "start_date": datetime.now(UTC()),
+            "is_cohorted": (course_cohort_settings.is_cohorted and
+                            entry["id"] in course_cohort_settings.cohorted_discussions)
+        }
```

Comment:
heads up on the changes in this file:  probably need to adjust since merging https://github.com/edx/edx-platform/pull/7081